### ISSUE TITLE
[Bug]: data race detected in bvt

### ISSUE THREAD

**Issue by jiangxinmeng1** [matrixorigin/matrixone#9537]
### Is there an existing issue for the same bug?

- [X] I have checked the existing issues.

### Environment

```Markdown
- Version or commit-id (e.g. v0.1.0 or 8b23a93):
- Hardware parameters:
- OS type:
- Others:
```

### Actual Behavior

![image](https://github.com/matrixorigin/matrixone/assets/51114574/8ebf19cb-a2ee-499d-85d3-313eab055f9e)

https://github.com/matrixorigin/matrixone/actions/runs/5012840276

[mo-service.log](https://github.com/matrixorigin/matrixone/files/11507283/mo-service.log)

**Comment by LeftHandCold** [comment]
@nnsgmsone plz.

**Comment by nnsgmsone** [comment]
fixed by #9711, @jiangxinmeng1
